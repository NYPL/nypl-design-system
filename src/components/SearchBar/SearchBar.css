.search-bar {
  display: flex;
  flex-flow: column nowrap;

  @media (--medium-break) {
    flex-flow: row nowrap;
  }

  .select {
    flex: 1 1 25%;
  }

  .button {
    margin-bottom: var(--space-xs);
  }
}
