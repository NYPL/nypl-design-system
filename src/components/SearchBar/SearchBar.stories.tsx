import * as React from "react";

import SearchBar from "./SearchBar";
import Select from "../Select/Select";
import HelperErrorText from "../HelperErrorText/HelperErrorText";
import { IconRotationTypes, IconNames } from "../Icons/IconTypes";
import Button from "../Button/Button";
import { ButtonTypes, ButtonIconPositions } from "../Button/ButtonTypes";
import Input from "../Input/Input";
import List from "../List/List";
import { ListTypes } from "../List/ListTypes";
import { InputTypes } from "../Input/InputTypes";
import { action } from "@storybook/addon-actions";
import { text, boolean, select } from "@storybook/addon-knobs";

export default {
    title: "SearchBar",
    component: SearchBar,
};

const optionsGroup = {
    Art: [
        "Academic Painting",
        "Amazing Painting",
        "Ancient Statue",
        "Basic Painting",
        "Beautiful Statue",
    ],
    Bushes: [
        "Blue-Hydrangea Bush",
        "Blue-Hydrangea Start",
        "Holly Bush",
        "Orange-Tea-Olive Bush",
        "Pink Camellia Bush",
    ],
    Clothing: ["3D Glasses", "Academy Uniform", "Acid-Washed Jacket"],
    "DIY Recipes": ["Acorn Pochette", "Acoustic Guitar", "Angled Signpost"],
    Flowers: [
        "Black Cosmos",
        "Blue Pansies",
        "Gold Roses",
        "Lily of the Valley",
        "Orange Hyacinth",
    ],
    Fossils: [
        "Acanthostega",
        "Amber",
        "Ammonite",
        "Ankylo Skull",
        "Anomalocaris",
        "Australopith",
    ],
    Fruits: ["Apples", "Oranges", "Pears", "Coconuts", "Cherries"],
    Furniture: [
        "Abstract Wall",
        "Accessories Stand",
        "ACNH Switch",
        "Air Circulator",
    ],
    Materials: [
        "Acorn",
        "Aquaris Fragment",
        "Bamboo Piece",
        "Boot",
        "Clump of Weeds",
    ],
    NPC: ["Celeste", "Kicks", "Leif", "Redd", "Saharah"],
    Songs: ["Agent K.K.", "CafÃ© K.K.", "Comrade K.K."],
    Tools: [
        "Axe",
        "Bamboo Wand",
        "Fishing Rod",
        "Shovel",
        "Slingshot",
        "Watering Can",
    ],
    Villagers: ["Agnes", "Alfonso", "Anchovy", "Axel", "Aurora"],
};

let showSelect, showHelperText, formErrored, formDisabled;

export const searchBar = () => (
    <>
        {boolean("Form Errored", false)
            ? (formErrored = true)
            : (formErrored = false)}
        {boolean("Form Disabled", false)
            ? (formDisabled = true)
            : (formDisabled = false)}
        {boolean("Show Select", true)
            ? (showSelect = true)
            : (showSelect = false)}
        {boolean("Show Helper Text", true)
            ? (showHelperText = true)
            : (showHelperText = false)}
        <SearchBar
            onSubmit={action("clicked")}
            ariaLabel={"New Horizons Item Search"}
        >
            {showSelect && (
                <Select
                    ariaLabel="Filter Search"
                    disabled={formDisabled}
                    helperTextId={"helperText"}
                    id={"select"}
                    isRequired={false}
                    labelId={"label"}
                    modifiers={formErrored ? ["error"] : null}
                    onBlur={action("blur")}
                    onChange={action("changed")}
                >
                    {Object.keys(optionsGroup).map((nhOption) => (
                        <option aria-selected={false}>{nhOption}</option>
                    ))}
                </Select>
            )}
            <Input
                ariaLabelledBy={"button helper-error-text"}
                disabled={formDisabled}
                errored={formErrored ? true : false}
                id="input1"
                placeholder={"Item Search"}
                required={true}
                type={InputTypes.text}
            ></Input>
            <Button
                buttonType={ButtonTypes.Primary}
                iconDecorative={true}
                iconName={IconNames.search_small}
                iconPosition={ButtonIconPositions.Left}
                id="button"
                onClick={action("clicked")}
                type="submit"
                disabled={formErrored || formDisabled ? true : false}
            >
                Search
            </Button>
        </SearchBar>
        {showHelperText && (
            <HelperErrorText
                id={"helper-error-text"}
                isError={formErrored ? true : false}
            >
                {formErrored
                    ? "Sorry, we can't find that item!"
                    : "Search for items in Animal Crossing New Horizons"}
            </HelperErrorText>
        )}
    </>
);

searchBar.story = {
    name: "Search Bar",
    parameters: {
        design: {
            type: "figma",
            url:
                "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=11689%3A423",
        },
    },
};
