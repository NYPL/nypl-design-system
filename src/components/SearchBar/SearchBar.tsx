import * as React from "react";
import bem from "../../utils/bem";

export interface SearchBarProps {
  /** Populates aria-label on the form;
   * defines a hidden string that labels
   * the SearchBar
   */
  ariaLabel?: string;
  /** Populates aria-labelledby on the form;
   * defines a visible string that labels
   * the SearchBar
   */
  ariaLabelledBy?: string;
  /** Additional attributes passed to the form */
  attributes?: { [key: string]: any };
  /** BlockName for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  blockName?: string;
  /** ClassName that appears in addition to "search-bar" */
  className?: string;
  /** Adds 'method' property to the form */
  method?: string;
  /** Adds 'action' property to the form */
  action?: string;
  /** ID that other components can cross reference for accessibility purposes */
  id?: string;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
  /** Handler on form submit */
  onSubmit: (event: React.FormEvent) => void;
}

/** Renders a wrapper `form` element to be used with `Select` (optional),
 * `Input`, and `Button` components together. */
export default function SearchBar(
  props: React.PropsWithChildren<SearchBarProps>
) {
  const {
    ariaLabel,
    ariaLabelledBy,
    blockName,
    children,
    className,
    id,
    modifiers = [],
    onSubmit,
    attributes,
    method,
    action,
  } = props;

  const baseClass = "search-bar";

  return (
    <form
      className={bem(baseClass, modifiers, blockName, [className])}
      id={id}
      role="search"
      aria-label={ariaLabel}
      aria-labelledby={ariaLabelledBy}
      onSubmit={onSubmit}
      method={method}
      action={action}
      {...attributes}
    >
      {children}
    </form>
  );
}
