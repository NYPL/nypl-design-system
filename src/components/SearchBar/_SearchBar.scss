.search-bar {
  display: flex;
  flex-flow: column nowrap;

  @include breakpoint($breakpoint-medium) {
    flex-flow: row nowrap;
  }

  .select {
    flex: 1 1 25%;
  }

  .button {
    @include space-stack-xs;
  }
}
