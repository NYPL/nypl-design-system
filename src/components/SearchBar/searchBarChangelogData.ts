/** This data is used to populate the ComponentChangelogTable component.
 *
 * date: string (when adding new entry during development, set value as "Prerelease")
 * version: string (when adding new entry during development, set value as "Prerelease")
 * type: "Bug Fix" | "New Feature" | "Update";
 * affects: array["Accessibility" | "Documentation" | "Functionality" | "Styles"];
 * notes: array (will render as a bulleted list, add one array element for each list element)
 */
import { ChangelogData } from "../../utils/ComponentChangelogTable";

export const changelogData: ChangelogData[] = [
  {
    date: "Prerelease",
    version: "Prerelease",
    type: "Update",
    affects: ["Functionality"],
    notes: ["Adds span element around button text to allow CSS selection."],
  },
  {
    date: "2024-06-20",
    version: "3.1.6",
    type: "Update",
    affects: ["Styles"],
    notes: ["Adds interaction tests for the Controls story."],
  },
  {
    date: "2024-05-23",
    version: "3.1.4",
    type: "Update",
    affects: ["Accessibility"],
    notes: [
      "Updates the internal `TextInput` component's aria-describedby value to include the `SearchBar`'s helper text ID.",
    ],
  },
  {
    date: "2024-03-14",
    version: "3.0.0",
    type: "Update",
    affects: ["Styles"],
    notes: ["Chakra 2.8 update."],
  },
  {
    date: "2023-12-07",
    version: "2.1.3",
    type: "Update",
    affects: ["Accessibility", "Documentation", "Styles"],
    notes: [
      "Updated the `headingText` prop to allow JSX to render custom heading elements for accessible heading hierarchy.",
      "Added a z-index on hover to the select icon so it no longer disappears.",
    ],
  },
];
