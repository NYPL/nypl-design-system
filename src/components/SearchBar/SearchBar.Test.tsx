import { expect } from "chai";
import { stub } from "sinon";
import * as Enzyme from "enzyme";
import * as React from "react";

import Button from "../Button/Button";
import { ButtonTypes } from "../Button/ButtonTypes";
import SearchBar from "./SearchBar";

describe("SearchBar", () => {
  let searchBarSubmit, buttonCallback;
  beforeEach(() => {
    searchBarSubmit = stub();
  });

  it("Renders SearchBar", () => {
    const wrapper = Enzyme.shallow(
      <SearchBar id="id" ariaLabel="searchbar" onSubmit={searchBarSubmit} />
    );
    expect(wrapper.find(".searchbar")).to.have.lengthOf(1);
    expect(wrapper.find(".searchbar").prop("aria-label")).to.equal("searchbar");
  });

  it("Search Bar calls call back on Submit ", () => {
    const mounted = Enzyme.mount(
      <SearchBar ariaLabel="searchBar" id="id" onSubmit={searchBarSubmit}>
        <Button
          buttonType={ButtonTypes.Primary}
          id="button"
          onClick={buttonCallback}
          type="submit"
        >
          Search
        </Button>
      </SearchBar>
    );
    mounted.find("Button").simulate("click");
    expect(searchBarSubmit.callCount).to.equal(1);
  });
});
