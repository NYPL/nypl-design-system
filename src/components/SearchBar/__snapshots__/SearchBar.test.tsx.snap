// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders the UI snapshot correctly 1`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label - Search for items in Animal Crossing New Horizons"
    className="css-1xdhyk6"
    id="searchbar-form-basic"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        className="chakra-input css-0"
        disabled={false}
        id="searchbar-textinput-basic"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-basic"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-basic"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
  <div
    aria-atomic={true}
    aria-live="off"
    className="css-1xdhyk6"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Search for items in Animal Crossing New Horizons",
      }
    }
    data-isinvalid={false}
    id="basic-helperText"
  />
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 2`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label - Search for items in Animal Crossing New Horizons"
    className="css-1xdhyk6"
    id="searchbar-form-withSelect"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        <div
          className="chakra-select__wrapper css-42b2qy"
        >
          <select
            aria-label="Select a category"
            className="chakra-select css-1ik61og"
            disabled={false}
            id="searchbar-select-withSelect"
            name="selectName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            value="Songs"
          >
            <option
              value="Art"
            >
              Art
            </option>
            <option
              value="Bushes"
            >
              Bushes
            </option>
            <option
              value="Clothing"
            >
              Clothing
            </option>
            <option
              value="Flowers"
            >
              Flowers
            </option>
            <option
              value="Fossils"
            >
              Fossils
            </option>
            <option
              value="Fruits"
            >
              Fruits
            </option>
            <option
              value="Furniture"
            >
              Furniture
            </option>
            <option
              value="Songs"
            >
              Songs
            </option>
            <option
              value="Tools"
            >
              Tools
            </option>
            <option
              value="Villagers"
            >
              Villagers
            </option>
          </select>
          <div
            className="chakra-select__icon-wrapper css-162mkon"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon chakra-select__icon css-1grhd2q"
              focusable={false}
              id="searchbar-select-withSelect-icon"
              role="presentation"
              style={
                Object {
                  "color": "currentColor",
                  "height": "1em",
                  "width": "1em",
                }
              }
              title="arrow icon"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                strokeWidth="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  strokeLinecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  strokeLinecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  strokeMiterlimit="10"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
      
    </div>
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        className="chakra-input css-0"
        disabled={false}
        id="searchbar-textinput-withSelect"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-withSelect"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-withSelect"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
  <div
    aria-atomic={true}
    aria-live="off"
    className="css-1xdhyk6"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Search for items in Animal Crossing New Horizons",
      }
    }
    data-isinvalid={false}
    id="withSelect-helperText"
  />
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 3`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-withoutHelperText"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        className="chakra-input css-0"
        disabled={false}
        id="searchbar-textinput-withoutHelperText"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-withoutHelperText"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-withoutHelperText"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 4`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-invalidState"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-invalid={true}
        aria-label="Item Search - There is an error related to this field."
        className="chakra-input css-0"
        disabled={false}
        id="searchbar-textinput-invalidState"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-invalidState"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-invalidState"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 5`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-disabledState"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        className="chakra-input css-0"
        disabled={true}
        id="searchbar-textinput-disabledState"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      disabled={true}
      id="searchbar-button-disabledState"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-disabledState"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 6`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-requiredState"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        aria-required={true}
        className="chakra-input css-0"
        disabled={true}
        id="searchbar-textinput-requiredState"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search (Required)"
        required={true}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      disabled={true}
      id="searchbar-button-requiredState"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-requiredState"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 7`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-noBrandButtonType"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        aria-required={true}
        className="chakra-input css-0"
        disabled={true}
        id="searchbar-textinput-noBrandButtonType"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search (Required)"
        required={true}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      disabled={true}
      id="searchbar-button-noBrandButtonType"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-noBrandButtonType"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 8`] = `
<div
  className="css-1xdhyk6"
>
  <h2
    className="chakra-heading css-1xdhyk6"
    id="withHeading-heading"
  >
    A Heading
  </h2>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-withHeading"
    onSubmit={[MockFunction]}
    role="search"
  >
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-withHeading"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-withHeading"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 9`] = `
<div
  className="css-1xdhyk6"
>
  <p
    className="chakra-text css-1xdhyk6"
  >
    A description
  </p>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-withDescription"
    onSubmit={[MockFunction]}
    role="search"
  >
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-withDescription"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-withDescription"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 10`] = `
<div
  className="css-1xdhyk6"
>
  <h2
    className="chakra-heading css-1xdhyk6"
    id="withHeadingAndDescription-heading"
  >
    A Heading
  </h2>
  <p
    className="chakra-text css-1xdhyk6"
  >
    A description
  </p>
  <form
    aria-label="SearchBar label"
    className="css-1xdhyk6"
    id="searchbar-form-withHeadingAndDescription"
    onSubmit={[MockFunction]}
    role="search"
  >
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-withHeadingAndDescription"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-withHeadingAndDescription"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 11`] = `
<div
  className="css-1xdhyk6"
>
  <form
    aria-label="SearchBar label - Search for items in Animal Crossing New Horizons"
    className="css-kle7zy"
    id="searchbar-form-chakra"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        className="chakra-input css-0"
        disabled={false}
        id="searchbar-textinput-chakra"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-chakra"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-chakra"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
  <div
    aria-atomic={true}
    aria-live="off"
    className="css-1xdhyk6"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Search for items in Animal Crossing New Horizons",
      }
    }
    data-isinvalid={false}
    id="chakra-helperText"
  />
</div>
`;

exports[`SearchBar renders the UI snapshot correctly 12`] = `
<div
  className="css-1xdhyk6"
  data-testid="props"
>
  <form
    aria-label="SearchBar label - Search for items in Animal Crossing New Horizons"
    className="css-1xdhyk6"
    id="searchbar-form-props"
    onSubmit={[MockFunction]}
    role="search"
  >
    <div
      className="css-1xdhyk6"
    >
      <input
        aria-label="Item Search"
        className="chakra-input css-0"
        disabled={false}
        id="searchbar-textinput-props"
        name="textInputName"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onFocus={[Function]}
        placeholder="Item Search "
        required={false}
        step={null}
        type="text"
      />
      
    </div>
    <button
      className="chakra-button css-1me3d5y"
      data-testid="button"
      id="searchbar-button-props"
      onClick={null}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="searchbar-icon-props"
        role="img"
        title="search icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Search
    </button>
  </form>
  <div
    aria-atomic={true}
    aria-live="off"
    className="css-1xdhyk6"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Search for items in Animal Crossing New Horizons",
      }
    }
    data-isinvalid={false}
    id="props-helperText"
  />
</div>
`;
